addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,451 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.km200.KM200ThingType;
+import org.openhab.binding.km200.internal.KM200ChannelTypeProvider;
+import org.openhab.binding.km200.internal.KM200CommObject;
+import org.openhab.binding.km200.internal.KM200SwitchProgramService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link KM200ThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Eckhardt - Initial contribution
+ */
+public class KM200ThingHandler extends BaseThingHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(KM200ThingHandler.class);
+    private static URI configDescriptionUriChannel;
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Sets.newHashSet(THING_TYPE_DHW_CIRCUIT,
+            THING_TYPE_HEATING_CIRCUIT, THING_TYPE_SOLAR_CIRCUIT, THING_TYPE_HEAT_SOURCE, THING_TYPE_SYSTEM_APPLIANCE,
+            THING_TYPE_SYSTEM_HOLIDAYMODES, THING_TYPE_SYSTEM_SENSOR, THING_TYPE_GATEWAY, THING_TYPE_NOTIFICATION,
+            THING_TYPE_SYSTEM);
+
+    private Boolean isInited = false;
+
+    private KM200ChannelTypeProvider channelTypeProvider;
+
+    public KM200ThingHandler(Thing thing, KM200ChannelTypeProvider channelTypeProvider) {
+        super(thing);
+        this.channelTypeProvider = channelTypeProvider;
+        thing.setStatusInfo(new ThingStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.NONE, ""));
+        try {
+            configDescriptionUriChannel = new URI(CONFIG_DESCRIPTION_URI_CHANNEL);
+        } catch (Exception ex) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
OK, done.